### PR TITLE
LPS-195794 Adding tests for Data Set Action with Object in the Data Set fragment

### PR DESCRIPTION
**Pull request by JerryAdrianoo**
**Description:**
Adding tests for Data Set Action with Object in the Data Set fragment 

**Jira Ticket:**
https://liferay.atlassian.net/browse/LPS-195794

### CODE REVIEW

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:stop

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:stop

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
cc @CarlosBrichete failures unrelated

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 1]
```diff
@@ -595,4 +1137,184 @@ definition {
...
+	test HeadlessActionKeyIsNotMandatory {
+		task ("And clicks on Create Item Action button") {
+			LexiconEntry.gotoAdd();
+		}
+
+		task ("Then the Headless Action Key field is not mandatory") {
+			AssertElementNotPresent(
+				key_fieldName = "Headless Action Key",
+				locator1 = "DataSet#MANDATORY_FIELD");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:1151]: To check a field is not mandatory you have to try to save it and check that you can save it. Checking the mandatory icon is not enough. Please add this verification to the test.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 2]
```diff
@@ -595,4 +1137,184 @@ definition {
...
+			AssertElementPresent(
+				key_title = "New Item Action",
+				locator1 = "Header#H2_TITLE");
+		}
+
+		task ("Then the Icon field is not mandatory") {
+			AssertElementNotPresent(
+				key_fieldName = "Icon",
+				locator1 = "DataSet#MANDATORY_FIELD");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:1192]: To check a field is not mandatory you have to try to save it and check that you can save it. Checking the mandatory icon is not enough. Please add this verification to the test.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 3]
```diff
@@ -595,4 +1137,184 @@ definition {
...
+				key_fieldLabel = "Type",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "Link");
+		}
+
+		task ("Then the Confirmation Message field is present") {
+			AssertElementPresent(
+				key_placeholder = "Add a message here.",
+				locator1 = "FrontendDataSet#PLACEHOLDER");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:1230]: Here you are just checking the field but not the localization. Please add the verification.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 4]
```diff
@@ -595,4 +1137,184 @@ definition {
...
+			AssertElementPresent(
+				key_title = "New Item Action",
+				locator1 = "Header#H2_TITLE");
+		}
+
+		task ("And the Label field is mandatory") {
+			AssertElementPresent(
+				key_fieldName = "Label",
+				locator1 = "DataSet#MANDATORY_FIELD");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:1250]: To check a field is mandatory you have to try to save it and check that you can save it. Checking the mandatory icon is not enough. Please add this verification to the test.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 5]
```diff
@@ -595,4 +1137,184 @@ definition {
...
+			AssertElementPresent(
+				key_title = "New Item Action",
+				locator1 = "Header#H2_TITLE");
+		}
+
+		task ("Then the Type field is mandatory") {
+			AssertElementPresent(
+				key_fieldName = "Type",
+				locator1 = "DataSet#MANDATORY_FIELD");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:1277]: To check a field is mandatory you have to try to save it and check that you can save it. Checking the mandatory icon is not enough. Please add this verification to the test.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 6]
```diff
@@ -595,4 +1137,184 @@ definition {
...
+	test MandatoryURLField {
+		task ("And clicks on plus button") {
+			LexiconEntry.gotoAdd();
+		}
+
+		task ("Then the URL field is mandatory") {
+			AssertElementPresent(
+				key_fieldName = "URL",
+				locator1 = "DataSet#MANDATORY_FIELD");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:1303]: To check a field is mandatory you have to try to save it and check that you can save it. Checking the mandatory icon is not enough. Please add this verification to the test.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase [hunk 7]
```diff
@@ -595,4 +1137,184 @@ definition {
...
+	test MessageTypeIsNotMandatory {
+		task ("And clicks on Create Item Action button") {
+			LexiconEntry.gotoAdd();
+		}
+
+		task ("Then the Message Type field is not mandatory") {
+			AssertElementNotPresent(
+				key_fieldName = "Message Type",
+				locator1 = "DataSet#MANDATORY_FIELD");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewActions.testcase:1317]: To check a field is not mandatory you have to try to save it and check that you can save it. Checking the mandatory icon is not enough. Please add this verification to the test.

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]
Please check my modification requests, thanks :)

**Comment by JerryAdrianoo** [comment]
Hey @CarlosBrichete, the changes you requested are related to another ticket about creating action. So, I rebased with master and now it is updated. Can I forward it?

**Review by CarlosBrichete — APPROVED** [review]
lgtm

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:forward

**Comment by JerryAdrianoo** [comment]
ci:stop

**Comment by JerryAdrianoo** [comment]
ci:stop

**Pull request closed without merging**